docs/pages/preview/responsive.tsx

```diff
@@ -0,0 +1,400 @@
+import { useIsIos } from '../../../packages/react/src/combobox/utils';
+
+const isLocal =
+	!process.env.NEXT_PUBLIC_BASE_PATH && process.env.NODE_ENV !== 'production';
+const trustedUrls = [
+	process.env.NEXT_PUBLIC_SITE_URL,
+	process.env.NEXT_PUBLIC_PLAYROOM_URL?.startsWith('/')
+		? undefined
+		: process.env.NEXT_PUBLIC_PLAYROOM_URL,
+].filter(Boolean);
+
+const screenSizes = {
+	xs: {
+		label: '320px (xs)',
+		width: 320,
+	},
+	sm: {
+		label: `${tokens.breakpoint.sm}px (sm)`,
+		width: tokens.breakpoint.sm,
+	},
+	md: {
+		label: `${tokens.breakpoint.md}px (md)`,
+		width: tokens.breakpoint.md,
+	},
+	lg: {
+		label: `${tokens.breakpoint.lg}px (lg)`,
+		width: tokens.breakpoint.lg,
+	},
+	xl: {
+		label: `${tokens.breakpoint.xl}px (xl)`,
+		width: tokens.breakpoint.xl,
+	},
+	xxl: {
+		label: `${tokens.breakpoint.xxl}px (xxl)`,
+		width: tokens.breakpoint.xxl,
+	},
+};
+type Sizes = keyof typeof screenSizes;
+
+const constructRadioId = (size: Sizes) => `radio-id-${size}`;
+
+type UnmountTargets = 'radio' | 'select';
+
+export default function ResponsivePage() {
+	const router = useRouter();
+	const pathname = usePathname();
+	const searchParams = useSearchParams();
+	const [frameSize, setFrameSize] = useState<Sizes>('xs');
+	const [iframeSrc, setIframeSrc] = useState<undefined | null | string>(
+		undefined
+	);
+	const iframeRef = useRef<HTMLIFrameElement>(null);
+	const selectRef = useRef<HTMLSelectElement>(null);
+	const scrollbarSizes = useGetScrollbarSizes();
+	const isIos = useIsIos();
+
+	useEffect(() => {
+		const initQueryParams = new URLSearchParams(window.location.search);
+		const defaultSize = initQueryParams.get(
+			responsivePreviewQueryKeys.previewSize
+		);
+
+		if (defaultSize && defaultSize in screenSizes) {
+			setFrameSize(defaultSize as Sizes);
+		}
+	}, [setFrameSize]);
+
+	useEffect(() => {
+		const initQueryParams = new URLSearchParams(window.location.search);
+		const iframeSrc = initQueryParams.get(responsivePreviewQueryKeys.frameSrc);
+
+		// Verify address is from AgDS
+		if (iframeSrc) {
+			try {
+				const encodedUrl = encodeURI(iframeSrc);
+
+				// If URL is AgDS, allow through
+				if (encodedUrl.startsWith('/')) {
+					setIframeSrc(encodedUrl);
+					return;
+				}
+
+				const parsedUrl = new URL(encodedUrl);
+				const isValidSrc = trustedUrls.some(
+					(domain) => domain && parsedUrl.hostname === new URL(domain).hostname
+				);
+				if (isValidSrc) {
+					setIframeSrc(encodedUrl);
+					return;
+				}
+			} catch {
+				setIframeSrc(null);
+				return;
+			}
+		}
+
+		// Invalid url or no url found, set as null
+		setIframeSrc(null);
+	}, [setIframeSrc]);
+
+	const referrerLink =
+		searchParams.get(responsivePreviewQueryKeys.referrerLink) || '/';
+	const referrerLabel = `Back to ${
+		searchParams.get(responsivePreviewQueryKeys.referrerLabel) ||
+		'documentation'
+	}`;
+	const title =
+		searchParams.get(responsivePreviewQueryKeys.title) || 'Responsive preview';
```

Comment:
It's annoying how this has a visible flash of content change, but not sure there's anything you can do :(